agent/builder.go

```diff
@@ -50,17 +61,23 @@ func (b *LogAgentBuilder) Build() (*LogAgent, error) {
 		return nil, errors.Wrap(err, "open database")
 	}
 
-	registry, err := plugin.NewPluginRegistry(b.pluginDir)
-	if err != nil {
-		return nil, errors.Wrap(err, "load plugin registry")
+	if b.pluginDir != "" {
+		if err := plugin.RegisterPlugins(b.pluginDir, operator.DefaultRegistry); err != nil {
+			return nil, err
+		}
 	}
 
-	buildContext := operator.BuildContext{
-		Logger:   b.logger,
-		Database: db,
+	if b.config != nil && len(b.configFiles) > 0 {
+		return nil, errors.NewError("agent can be built WithConfig or WithConfigFiles, but not both", "")
```

Comment:
What happens if neither `WithConfig` or `WithConfigFiles` has been called? Doesn't look like we handle that condition. Should we?

Reply:
Good call. We should